javalib/src/main/scala/java/nio/channels/FileChannelImpl.scala

```diff
@@ -144,18 +144,21 @@ private object FileChannelImpl {
     val mode = new StringBuilder("r")
     if (options.contains(WRITE) || options.contains(APPEND)) mode.append("w")
 
-    if (!Files.exists(path, Array.empty) && !options.contains(CREATE) && !options
-          .contains(CREATE_NEW)) {
-      throw new NoSuchFileException(path.toString)
-    }
-
-    if (options.contains(CREATE_NEW) && Files.exists(path, Array.empty)) {
-      throw new FileAlreadyExistsException(path.toString)
-    }
-
-    if (options.contains(WRITE) && (options.contains(CREATE_NEW) || options
-          .contains(CREATE))) {
-      Files.createFile(path, attrs)
+    if (!Files.exists(path, Array.empty)) {
+      if (!options.contains(CREATE) && !options.contains(CREATE_NEW)) {
+        throw new NoSuchFileException(path.toString)
+      } else if (options.contains(WRITE)) {
+        Files.createFile(path, attrs)
+      }
```

Comment:
In 2., we will throw an exception if `options` doesn't contain `CREATE` and `CREATE_NEW`. I changed that to make it clearer.

Regarding the spec, my reading is that we shouldn't try to create the file if we don't open it for writing:
> `CREATE_NEW`: This option is ignored when the file is opened only for reading. 

> `CREATE`: This option is ignored if the CREATE_NEW option is also present or the file is opened only for reading. 

My understanding is also that if none of `WRITE` or `APPEND` is specified, then the file should be opened for reading only:

> The READ and WRITE options determine if the file should be opened for reading and/or writing. If neither option (or the APPEND option) is contained in the array then the file is opened for reading.

I also fixed an issue with APPEND that didn't work correctly.